packages/x-adapter-platform/src/schemas/facets/utils.ts

```diff
@@ -1,12 +1,26 @@
-import { facetsConfig } from '../models/facet.schema';
+import { PlatformFacetType } from '../../types';
+import { hierarchicalFilterSchema, numberFilterSchema, simpleFilterSchema } from '../models';
 import { FacetConfig } from './types';
 
 /**
  * Returns the facet's config.
  *
- * @param facet - The facet to resolve the configuration.
+ * @param type - The facet type to resolve the configuration.
  * @returns The facet's config.
  */
-export function getFacetConfig(facet: string): FacetConfig {
-  return facetsConfig[facet] ?? facetsConfig.default;
+export function getFacetConfig(type: PlatformFacetType): FacetConfig {
+  return {
+    value: {
+      modelName: 'SimpleFacet',
+      schema: simpleFilterSchema
+    },
+    hierarchical: {
+      modelName: 'HierarchicalFacet',
+      schema: hierarchicalFilterSchema
+    },
+    range: {
+      modelName: 'NumberRangeFacet',
+      schema: numberFilterSchema
+    }
+  }[type] as FacetConfig;
```

Comment:
I'm wondering whether we need a defaulf fallback here or not to cover the case where we'd get from the service an unsupported type. I don't think so because in the end it is a standard API and we should plan and coorinate back & front for a new development in case a new type is needed, but happy to read other opinions :)

Reply:
I am not sure about this, I think that maybe it's a good idea but as you said if it's a standard API this should always be right.  I am not 100% sure that this is going to be right always, you know 😂

Reply:
Added just in case.